### PR TITLE
Convert remaining string literal .format to f-strings

### PR DESCRIPTION
**Pull request by peterjc**
This pull request addresses issue #2510, building on #2524 which only did the easiest cases with flynt.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/motifs/transfac.py [hunk 1]
```diff
@@ -193,24 +183,20 @@ def read(handle, strict=True):
...
-                    "should start with a "
-                    '"[": "{0:s}"'.format(index, line)
+                    f'The index "{index}" in a TRANSFAC RN line should start'
+                    f' with a "[": "{line}"'
                 )
             if index[-1] != "]":
                 raise ValueError(
-                    'The index "{0:s}" in a TRANSFAC RN line '
-                    "should end with a "
-                    '"]": "{0:s}"'.format(index, line)
```
Comment (peterjc) [inline Bio/motifs/transfac.py]: This example had a bug, ``{0:s}`` instead of ``{1:s}`` - a perfect example of a downside of this syntax.

**Comment by peterjc** [comment]
@chris-rands you might like to review this too?

File: Bio/SeqFeature.py [hunk 2]
```diff
@@ -445,10 +445,10 @@ def translate(
 
         if start_offset not in [0, 1, 2]:
             raise ValueError(
-                "The start_offset must be 0, 1, or 2. "
-                "The supplied value is {}. Check the value "
-                "of either the codon_start qualifier or "
-                "the start_offset argument".format(start_offset)
+                f"The start_offset must be 0, 1, or 2. "
```
Comment (chris-rands) [inline Bio/SeqFeature.py]: `f"` is redundant on this line. Also a minor style point, should it be aligned like this:
```
f"a"
"b"
```
or
```
f"a"
 "b"
```

File: Bio/__init__.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ class BiopythonExperimentalWarning(BiopythonWarning):
         " missing compiled C extensions. We recommend that you"
         " try running your code from outside the source tree."
         " If you are outside the source tree then you have a"
-        " setup.py file in an unexpected directory: {}.".format(_parent_dir),
+        " setup.py file in an unexpected directory: " + _parent_dir,
```
Comment (chris-rands) [inline Bio/__init__.py]: could (optionally) be an f-string

File: Bio/SeqFeature.py (hunk 2 above)
Comment (peterjc) [reply Bio/SeqFeature.py]: Good catch, thank you - there was a flake8 plugin which caught this kind of pointless f-string, but it had problems with false positives.

The black alignment is the first of the two.

File: Bio/__init__.py (hunk 3 above)
Comment (peterjc) [reply Bio/__init__.py]: It could - honestly unsure in a case like this which is faster, or which is clearer.

**Pull request merged**